dashboard/modules/log/log_head.py

```diff
@@ -22,23 +22,27 @@ def __init__(self, dashboard_head):
             self.insert_log_url_to_node_info)
 
     async def insert_log_url_to_node_info(self, node_info):
-        ip = node_info.get("ip")
-        if ip is None:
+        node_id = node_info.get("raylet", {}).get("nodeId")
+        if node_id is None:
             return
-        agent_port = DataSource.agents.get(ip)
+        agent_port = DataSource.agents.get(node_id)
         if agent_port is None:
             return
         agent_http_port, _ = agent_port
-        log_url = self.LOG_URL_TEMPLATE.format(ip=ip, port=agent_http_port)
+        log_url = self.LOG_URL_TEMPLATE.format(
```

Comment:
`node info` is a combination of `node physical stats` *from reporter agent*, `node stats` *from raylet* and `GcsNodeInfo` *from GCS*. When a node info query arrives, some data may not be ready, for example, the datacenter has not received the reporter data. So, I use `get(node_id)` instead of `[node_id]` here.